### PR TITLE
fix bees runtime

### PR DESCRIPTION
**Pull request by gurfan**
hopefully

### CODE REVIEW

**Comment by DamianX** [comment]
why does the `bees` list contain null, though?

**Comment by Zth--** [comment]
Is null a valid value for that list?

**Comment by gurfan** [comment]
var/datum/bee/B is ending up as null somehow and causing a runtime

**Comment by Exxion** [comment]
I suspect a bee in that list is getting hard deleted and leaving a null in there (because apparently that's what happens when something in a list is hard deleted)

**Comment by DamianX** [comment]
No shit. The question is why! That sounds like it could be related to a hard del.

File: code/modules/mob/living/simple_animal/bees/bees_datums.dm [hunk 1]
```diff
@@ -55,6 +55,7 @@ var/bees_count = 0
 
 /datum/bee/proc/death(var/gibbed = FALSE)
 	if (mob)
+		mob.bees.Remove(src)
```
Comment (Exxion) [inline code/modules/mob/living/simple_animal/bees/bees_datums.dm]: This already happens in `Destroy()`, so this won't do anything

**Comment by gurfan** [comment]
If the bee datum belongs to a mob now, it'll get removed from the bees list when it dies. I think this will fix bees qdel'd while I list still references them

File: code/modules/mob/living/simple_animal/bees/bees_datums.dm (hunk 1 above)
Comment (gurfan) [reply code/modules/mob/living/simple_animal/bees/bees_datums.dm]: gah

Comment (gurfan) [reply code/modules/mob/living/simple_animal/bees/bees_datums.dm]: I will keep looking around

**Comment by gurfan** [comment]
I will reopen this when I take a better look at it, there's lots to fix

**Comment by Exxion** [comment]
After a bunch of searching, I believe this is caused by the bug net. It adds all the bees in a mob to its list of caught bees then deletes the mob, but does not remove the bees from the mob before deleting it, resulting in the bees in the net having been `qdel()`ed but still referenced by the net. If the net is then emptied before the GC hard dels those bees, they'll be put into a new mob, where they will eventually be hard deleted, leaving nulls in the list.
The solution seems to be to just remove the bees from the bee mob when the bug net picks them up.

**Comment by gurfan** [comment]
I've made it so they're removed from the list. I looked over all the bee items and I made the mistake of assuming that the bee mob would handle everything correctly when it's destroyed since it clears all bees from it's list in destroy(), I didn't think that the net would be able to empty before the bee mob is qdel'd

**Pull request merged**